test/ec_internal_test.c

```diff
@@ -183,6 +183,47 @@ static int field_tests_default(int n)
     return ret;
 }
 
+static
+int ecpkparams_i2d2i_test(int n)
+{
+    EC_GROUP *g1 = NULL, *g2 = NULL;
+    FILE *fp = NULL;
+    int nid = curves[n].nid;
+    int testresult = 0;
+
+    /* create group */
+    if (!TEST_ptr(g1 = EC_GROUP_new_by_curve_name(nid)))
+        goto end;
+
+    /* encode params to file */
+    if (!TEST_ptr(fp = fopen("params.der", "wb"))
+            || !TEST_true(i2d_ECPKParameters_fp(fp, g1)))
+        goto end;
+
+    /* flush and close file */
+    if (!TEST_int_eq(fclose(fp), 0)) {
```

Comment:
I assume windoze is complaining about one of the fopen/fclose calls.. maybe this check here?

Reply:
@slontis do you have a suggestion on how to address it?

Reply:
Reading the logs indicates it an applink related problem..
For a semi-related issue see https://github.com/openssl/openssl/issues/10086.
Maybe try using BIO instead and see if it disappears.

Reply:
@romen ?

Reply:
Sorry I lost track of this.

So BIO is a no go, because the whole point of this PR is to add tests to cover the fp versions that are currently not tested.

So in windows it is not possible to use fopen and friends at all?
Should I just disable the new tests in windows? (what kind of defines could I use for that?)

Reply:
If you looks at the tricks that the bio fp does to get the fp to work you could do it.. otherwise disable it.

Reply:
it's a regression test so IMO it is perfectly fine to just disable it on Windows. If someone uses it on Windows and it breaks again, they can fix it.

Reply:
The applink/uplink stuff ensures that libcrypto understands FILE the way the application that links with it does, not the way the C RTL libcrypto was linked with does. This is crucial when libcrypto is linked with the debugging C RTL and the application is not.

This wouldn't be necessary if BIO was a more complete IO library.

So you must build every program that passes a FILE to be wrapped in a BIO with `ms/app link.c` on Windows. The beginning of `apps/build.info` shows how to set up a practical variable for it.

We may have been lucky with some programs...

Reply:
@levitte I can see something related to this in the apps/build.info in master, but not in 1.1.1.

How is this problem solved in 1.1.1?

Should this PR fix the Windows build for all the tests or only for this one?

Reply:
Ping @levitte